rustls/src/conn.rs

```diff
@@ -588,7 +588,14 @@ impl<Data> ConnectionCommon<Data> {
             }
 
             while self.wants_write() {
-                wrlen += self.write_tls(io)?;
+                match self.write_tls(io)? {
```

Comment:
Commenting here because it isn't possible to leave feedback on the commit message directly:

There's a typo in your commit title: "Avoid hagning if the client disconnects" -> "Avoid *hanging* if the client disconnects"